### ISSUE TITLE
"Unfortunately, Owncloud has stopped" after photo when AutoUpload is enabled

### ISSUE THREAD

**Issue by domneu** [owncloud/android#1341]
After a photo is taken, I get the above behavior and the photo never gets transferred to the server.  This occurs over cellular, as well as wifi.  It even happens when wifi-only is enabled and the phone is not connected to wifi.  This is only occurring on our 50 or so Verizon Kyocera Brigadier phones.   Our other phones (mostly Motorolas) work fine.  If there is anything I can do to help solve this, please let me know.

domneu

**Comment by tobiasKaminsky** [comment]
A logcat would be fine, otherwise we cannot see where the problem is.

**Comment by domneu** [comment]
It is not rooted.   Is a logcat still possible?

**Comment by tobiasKaminsky** [comment]
Of course: https://play.google.com/store/apps/details?id=com.nolanlawson.logcat
Then use "owncloud" to filter your logcat and please post the output right after an unsuccessful instant upload.

**Comment by domneu** [comment]
Sorry, I have done what you requested, but the logcat is blank when I
filter by owncloud.  Again, I do NOT have root on this phone.  Is this the
expected bahavior?

Thank You,
Dominic

**Comment by przybylski** [comment]
Yes, since 4.2 this app requires root.
You can try to reproduce this issue on beta, if this still happens then you can provide logs to us which is stored on the device directly.

**Comment by domneu** [comment]
I'm attempting the beta route.  Where are the logs in the beta version
stored?

**Comment by domneu** [comment]
************ CAUSE OF ERROR ************

java.lang.RuntimeException: Unable to start receiver
com.owncloud.android.files.InstantUploadBroadcastReceiver:
java.lang.NullPointerException
at android.app.ActivityThread.handleReceiver(ActivityThread.java:2430)
at android.app.ActivityThread.access$1700(ActivityThread.java:139)
at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1276)
at android.os.Handler.dispatchMessage(Handler.java:102)
at android.os.Looper.loop(Looper.java:136)
at android.app.ActivityThread.main(ActivityThread.java:5057)
at java.lang.reflect.Method.invokeNative(Native Method)
at java.lang.reflect.Method.invoke(Method.java:515)
at
com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:786)
at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:602)
at dalvik.system.NativeStart.main(Native Method)
Caused by: java.lang.NullPointerException
at
android.content.ContentResolver.acquireUnstableProvider(ContentResolver.java:1420)
at android.content.ContentResolver.query(ContentResolver.java:445)
at android.content.ContentResolver.query(ContentResolver.java:404)
at
com.owncloud.android.files.InstantUploadBroadcastReceiver.handleNewPictureAction(InstantUploadBroadcastReceiver.java:99)
at
com.owncloud.android.files.InstantUploadBroadcastReceiver.onReceive(InstantUploadBroadcastReceiver.java:69)
at android.app.ActivityThread.handleReceiver(ActivityThread.java:2423)
... 10 more

************ DEVICE INFORMATION ***********
Brand: kyocera
Device: E6782
Model: E6782
Id: KVT49L
Product: E6782

************ FIRMWARE ************
SDK: 19
Release: 4.4.2
Incremental: 1031_2045

**Comment by domneu** [comment]
Is this information helpful, or do you need something more?

**Comment by tobiasKaminsky** [comment]
This is helpful, but without trying to debug/analyse the code I cannot say for now which is the problem...

**Comment by davivel** [comment]
Tracking the stack trace, the problem seems that our line 
https://github.com/owncloud/android/blob/master/src/com/owncloud/android/files/InstantUploadBroadcastReceiver.java#L98 
is passing a NULL value in the first parameter. That results in a crash in
https://github.com/owncloud/android/blob/master/src/com/owncloud/android/files/InstantUploadBroadcastReceiver.java#L98 .

Protecting the code of the crash is easy, but the instant upload will not work in those devices, since for some reason they are sending a Intent with an unexpected content.

@domenu, I can send a fix to beta so that you may test by yourself.

**Comment by davivel** [comment]
Added a new log, maybe helps to know why the URI is NULL.

**Comment by domneu** [comment]
Thank you.  I will try it a bit later and reply with the results.

**Comment by AndyScherzinger** [comment]
I can also test it on the beta branch during the next days since this also happens on my Nexus5X running Stock 6.0.1

**Comment by domneu** [comment]
Ok, I have a dumb question... How do I download the latest beta .apk?  It seems that I have to download the source and then compile it.  Is this correct?  Or is it hosted somewhere already compiled?  The one on f-droid is a few days old.  Sorry about this.

**Comment by przybylski** [comment]
@domneu .apk is made by @tobiasKaminsky from time to time and uploaded to f-droid, if you want to have latest stuff you would need to compile code yourself

**Comment by tobiasKaminsky** [comment]
I will do a new beta on sunday. 
They are also listed in folder "apk" on branch "beta".

**Comment by tobiasKaminsky** [comment]
Beta is now available here: 
https://github.com/owncloud/android/tree/beta/apks

The app also has a manual update check in settings

**Comment by domneu** [comment]
For some reason, the Instant Upload of photos would not work.  Owncloud
would not error or attempt to upload any photos.  I tried clearing cache,
Wi-Fi only, clearing app data, etc., but nothing worked.  Finally, I tried
taking a video and got this error log:

************ CAUSE OF ERROR ************

java.lang.RuntimeException: Unable to start receiver
com.owncloud.android.files.InstantUploadBroadcastReceiver:
java.lang.NullPointerException
at android.app.ActivityThread.handleReceiver(ActivityThread.java:2430)
at android.app.ActivityThread.access$1700(ActivityThread.java:139)
at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1276)
at android.os.Handler.dispatchMessage(Handler.java:102)
at android.os.Looper.loop(Looper.java:136)
at android.app.ActivityThread.main(ActivityThread.java:5057)
at java.lang.reflect.Method.invokeNative(Native Method)
at java.lang.reflect.Method.invoke(Method.java:515)
at
com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:786)
at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:602)
at dalvik.system.NativeStart.main(Native Method)
Caused by: java.lang.NullPointerException
at
android.content.ContentResolver.acquireUnstableProvider(ContentResolver.java:1420)
at android.content.ContentResolver.query(ContentResolver.java:445)
at android.content.ContentResolver.query(ContentResolver.java:404)
at
com.owncloud.android.files.InstantUploadBroadcastReceiver.handleNewVideoAction(InstantUploadBroadcastReceiver.java:176)
at
com.owncloud.android.files.InstantUploadBroadcastReceiver.onReceive(InstantUploadBroadcastReceiver.java:74)
at android.app.ActivityThread.handleReceiver(ActivityThread.java:2423)
... 10 more

************ DEVICE INFORMATION ***********
Brand: kyocera
Device: E6782
Model: E6782
Id: KVT49L
Product: E6782

************ FIRMWARE ************
SDK: 19
Release: 4.4.2
Incremental: 1031_2045

**Comment by przybylski** [comment]
@domneu it looks like it's uploading video, not picture, the same fix, as for pictures, needs to be applied as provided by @davivel few days ago

**Comment by davivel** [comment]
True, I missed the flow of video uploads. Shame on me. Will fix it.

Anyway, this is what I expected for pictures:

> the instant upload will not work in those devices, since for some reason they are sending a Intent with an unexpected content.

Mmm, I added an extra line to log, but I thought that beta APKs had the logs entry enabled in the navigation drawer. Now I see that's not the case. 

I'll make a new PR.

@domneu , meanwhile, could you check if the Google camera app may be installed in those devices? Could you check if the instant upload of pictures works with it? You can find it in https://play.google.com/store/apps/details?id=com.google.android.GoogleCamera

**Comment by tobiasKaminsky** [comment]
> Mmm, I added an extra line to log, but I thought that beta APKs had the logs entry enabled in the navigation drawer. Now I see that's not the case. 

I can enable it.

**Comment by tobiasKaminsky** [comment]
For me there is "Logs" in the drawer?

**Comment by davivel** [comment]
Not in the version in F-Droid, that's available for beta users. 

That entry is conditioned to be available only in debug APKs; the F-Droid version is in release mode, I guess.

**Comment by tobiasKaminsky** [comment]
Right. I had only checked it in my local debug version.
Now I have changed it and it should be available in f-droid tomorrow.

**Comment by domneu** [comment]
great. I will download it again and report the results.

**Comment by AndyScherzinger** [comment]
I also have this problem on my Nexus5X running Android 6.0.1, just the beta (installed via build in Android Studio), no "standard" oC version in parallel. The permission error seems weird since it seems I am the only one with this problem:

************ CAUSE OF ERROR ************

java.lang.RuntimeException: Unable to start receiver com.owncloud.android.files.InstantUploadBroadcastReceiver: java.lang.SecurityException: Permission Denial: reading com.android.providers.media.MediaProvider uri content://media/external/images/media/35719 from pid=14277, uid=10156 requires android.permission.READ_EXTERNAL_STORAGE, or grantUriPermission()
at android.app.ActivityThread.handleReceiver(ActivityThread.java:2732)
at android.app.ActivityThread.-wrap14(ActivityThread.java)
at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1421)
at android.os.Handler.dispatchMessage(Handler.java:102)
at android.os.Looper.loop(Looper.java:148)
at android.app.ActivityThread.main(ActivityThread.java:5417)
at java.lang.reflect.Method.invoke(Native Method)
at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:726)
at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:616)
Caused by: java.lang.SecurityException: Permission Denial: reading com.android.providers.media.MediaProvider uri content://media/external/images/media/35719 from pid=14277, uid=10156 requires android.permission.READ_EXTERNAL_STORAGE, or grantUriPermission()
at android.os.Parcel.readException(Parcel.java:1620)
at android.database.DatabaseUtils.readExceptionFromParcel(DatabaseUtils.java:183)
at android.database.DatabaseUtils.readExceptionFromParcel(DatabaseUtils.java:135)
at android.content.ContentProviderProxy.query(ContentProviderNative.java:421)
at android.content.ContentResolver.query(ContentResolver.java:491)
at android.content.ContentResolver.query(ContentResolver.java:434)
at com.owncloud.android.files.InstantUploadBroadcastReceiver.handleNewPictureAction(InstantUploadBroadcastReceiver.java:105)
at com.owncloud.android.files.InstantUploadBroadcastReceiver.onReceive(InstantUploadBroadcastReceiver.java:70)
at android.app.ActivityThread.handleReceiver(ActivityThread.java:2725)
... 8 more

************ DEVICE INFORMATION ***********
Brand: google
Device: bullhead
Model: Nexus 5X
Id: MMB29K
Product: bullhead

************ FIRMWARE ************
SDK: 23
Release: 6.0.1
Incremental: 2419427

**Comment by davivel** [comment]
@AndyScherzinger , different issue, though maybe it's already registered something similar. Seems that we'll have to give some love to new permissions system in the end...

**Comment by tobiasKaminsky** [comment]
This could be #1343

**Comment by AndyScherzinger** [comment]
Probably the same issue as referenced by @tobiasKaminsky which would mean it is a Android 6 only problem (?) or is anyone experiencing the same issue on a pre Android 6?

**Comment by AndyScherzinger** [comment]
I deployed my fix to the beta branch and the originating marshmallow branch. This fixes the READ_EXTERNAL_STORAGE exception for the beta version :grinning: 
This also means I implemented the new permissions API introduced with Android M in a backwards compatible (yet untested) way for this permission (write permission). Relevant commit can be checked out here: https://github.com/owncloud/android/commit/373235225780503555388bc4a099f3c517a8129a

**Comment by tobiasKaminsky** [comment]
@AndyScherzinger To better keep track of the merges please do commits only to PRs and then we merge them into beta.
With this we can be sure that we do not merge commits into beta without having them in master.

(I have a script which shows me if a branch is "newer" than the merge state in beta, so I can easily merge the changed branches).

**Comment by tobiasKaminsky** [comment]
[Maybe it would also be better to have this as a separate PR as it is easier to track and review]

**Comment by domneu** [comment]
Let me know when these new changes are on f-droid and I will see if they
have fixed the problem on my Kyocera.  Thank you all for the help.

**Comment by AndyScherzinger** [comment]
@tobiasKaminsky I did the permission changes on the marshmallow branch and then pulled them to beta :smiley: - no worries :+1:

**Comment by arnottcr** [comment]
I am getting the same Cause of Error with the following device and firmware:

************ DEVICE INFORMATION ***********
Brand: oneplus
Device: A0001
Model: A0001
Id: MMB29T
Product: bacon

************ FIRMWARE ************
SDK: 23
Release: 6.0.1
Incremental: 20c8361ae4

**Comment by tobiasKaminsky** [comment]
I have updated the beta.
Can you please retest? @AndyScherzinger @arnottcr

**Comment by davivel** [comment]
People, we are mixing two different topics here. Could we focus back in the original issue, please?

@domneu, is the app still broken for you? Could you share with us some new logs with the fail, parting from the beta version in F-Droid?

Thanks in advance.

**Comment by domneu** [comment]
I won't be able to check it for a week.  I'm out of town.

**Comment by domneu** [comment]
I tried the latest beta from Fdroid.  The camera is not uploading
automatically at all.  No error, no transfer, no nothing.  When I  switch
on auto video upload, I get the following error:

************ CAUSE OF ERROR ************



java.lang.RuntimeException: Unable to start receiver
com.owncloud.android.files.InstantUploadBroadcastReceiver:
java.lang.NullPointerException

                at
android.app.ActivityThread.handleReceiver(ActivityThread.java:2430)

                at
android.app.ActivityThread.access$1700(ActivityThread.java:139)

                at
android.app.ActivityThread$H.handleMessage(ActivityThread.java:1276)

                at android.os.Handler.dispatchMessage(Handler.java:102)

                at android.os.Looper.loop(Looper.java:136)

                at android.app.ActivityThread.main(ActivityThread.java:5057)

                at java.lang.reflect.Method.invokeNative(Native Method)

                at java.lang.reflect.Method.invoke(Method.java:515)

                at
com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:786)

                at
com.android.internal.os.ZygoteInit.main(ZygoteInit.java:602)

                at dalvik.system.NativeStart.main(Native Method)

Caused by: java.lang.NullPointerException

                at
android.content.ContentResolver.acquireUnstableProvider(ContentResolver.java:1420)

                at
android.content.ContentResolver.query(ContentResolver.java:445)

                at
android.content.ContentResolver.query(ContentResolver.java:404)

                at
com.owncloud.android.files.InstantUploadBroadcastReceiver.handleNewVideoAction(InstantUploadBroadcastReceiver.java:187)

                at
com.owncloud.android.files.InstantUploadBroadcastReceiver.onReceive(InstantUploadBroadcastReceiver.java:76)

                at
android.app.ActivityThread.handleReceiver(ActivityThread.java:2423)

                ... 10 more



************ DEVICE INFORMATION ***********

Brand: kyocera

Device: E6782

Model: E6782

Id: KVT49L

Product: E6782



************ FIRMWARE ************

SDK: 19

Release: 4.4.2

Incremental: 1031_2045

**Comment by davivel** [comment]
Sorry, @domneu , I should have clarified.

Could you paste the log file right after taking a picture with instant upload enabled? Even though there is not result visible in the app, the log should be dumping some more information about what the app receives than before.

Thanks in advance.

**Comment by domneu** [comment]
Now I understand.  Thank you for your help on this.

Here it is:

************ CAUSE OF ERROR ************



java.lang.RuntimeException: Unable to start activity
ComponentInfo{com.owncloud.android.beta/com.owncloud.android.ui.activity.Uploader}:
java.lang.IndexOutOfBoundsException: Invalid index 0, size is 0

                at
android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2199)

                at
android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:2249)

                at
android.app.ActivityThread.access$800(ActivityThread.java:139)

                at
android.app.ActivityThread$H.handleMessage(ActivityThread.java:1200)

                at android.os.Handler.dispatchMessage(Handler.java:102)

                at android.os.Looper.loop(Looper.java:136)

                at android.app.ActivityThread.main(ActivityThread.java:5057)

                at java.lang.reflect.Method.invokeNative(Native Method)

                at java.lang.reflect.Method.invoke(Method.java:515)

                at
com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:786)

                at
com.android.internal.os.ZygoteInit.main(ZygoteInit.java:602)

                at dalvik.system.NativeStart.main(Native Method)

Caused by: java.lang.IndexOutOfBoundsException: Invalid index 0, size is 0

                at
java.util.ArrayList.throwIndexOutOfBoundsException(ArrayList.java:255)

                at java.util.ArrayList.get(ArrayList.java:308)

                at
com.owncloud.android.ui.activity.Uploader.somethingToUpload(Uploader.java:528)

                at
com.owncloud.android.ui.activity.Uploader.setAccount(Uploader.java:169)

                at
com.owncloud.android.ui.activity.FileActivity.onCreate(FileActivity.java:207)

                at
com.owncloud.android.ui.activity.Uploader.onCreate(Uploader.java:160)

                at android.app.Activity.performCreate(Activity.java:5231)

                at
android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1087)

                at
android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2163)

                ... 11 more



************ DEVICE INFORMATION ***********

Brand: kyocera

Device: E6782

Model: E6782

Id: KVT49L

Product: E6782



************ FIRMWARE ************

SDK: 19

Release: 4.4.2

Incremental: 1031_2045

**Comment by domneu** [comment]
Did the last log I sent help at all?

Thank you,
Dominic

**Comment by davivel** [comment]
No, sorry, that log is of a different crash triggered from a manual upload.

Try again, please. Do not search for a crash in the log, the information to see is that; instead, search for the word  "InstantUploadBroadcastReceiver".

**Comment by domneu** [comment]
Ok.  I will be able to try Monday.  Thank you for the help.

Dominic

**Comment by tobiasKaminsky** [comment]
@domneu any update?

**Comment by budinm1** [comment]
I believe I have the same problem as @domneu. Anyway, I have Asus Zenfone 2 with Cyanogenmod 13 beta (Android 6.0.1). Integrated Camera app have no problem with ownCloud autoupload. OwnCloud's auto upload have problem with Camera FV-5 (https://play.google.com/store/apps/details?id=com.flavionet.android.camera.pro). I have installed the latest beta, it didn't help.

`

************ CAUSE OF ERROR ************

java.lang.RuntimeException: Unable to start receiver com.owncloud.android.files.InstantUploadBroadcastReceiver: java.lang.SecurityException: Permission Denial: opening provider com.android.externalstorage.ExternalStorageProvider from ProcessRecord{277bc9b 24750:com.owncloud.android.beta/u0a198} (pid=24750, uid=10198) requires android.permission.MANAGE_DOCUMENTS or android.permission.MANAGE_DOCUMENTS
	at android.app.ActivityThread.handleReceiver(ActivityThread.java:2766)
	at android.app.ActivityThread.-wrap14(ActivityThread.java)
	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1440)
	at android.os.Handler.dispatchMessage(Handler.java:102)
	at android.os.Looper.loop(Looper.java:148)
	at android.app.ActivityThread.main(ActivityThread.java:5461)
	at java.lang.reflect.Method.invoke(Native Method)
	at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:726)
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:616)
	at de.robv.android.xposed.XposedBridge.main(XposedBridge.java:117)
Caused by: java.lang.SecurityException: Permission Denial: opening provider com.android.externalstorage.ExternalStorageProvider from ProcessRecord{277bc9b 24750:com.owncloud.android.beta/u0a198} (pid=24750, uid=10198) requires android.permission.MANAGE_DOCUMENTS or android.permission.MANAGE_DOCUMENTS
	at android.os.Parcel.readException(Parcel.java:1620)
	at android.os.Parcel.readException(Parcel.java:1573)
	at android.app.ActivityManagerProxy.getContentProvider(ActivityManagerNative.java:3593)
	at android.app.ActivityThread.acquireProvider(ActivityThread.java:4822)
	at android.app.ContextImpl$ApplicationContentResolver.acquireUnstableProvider(ContextImpl.java:2060)
	at android.content.ContentResolver.acquireUnstableProvider(ContentResolver.java:1473)
	at android.content.ContentResolver.query(ContentResolver.java:477)
	at android.content.ContentResolver.query(ContentResolver.java:435)
	at com.owncloud.android.files.InstantUploadBroadcastReceiver.handleNewPictureAction(InstantUploadBroadcastReceiver.java:121)
	at com.owncloud.android.files.InstantUploadBroadcastReceiver.onReceive(InstantUploadBroadcastReceiver.java:69)
	at android.app.ActivityThread.handleReceiver(ActivityThread.java:2759)
	... 9 more

************ DEVICE INFORMATION ***********
Brand: asus
Device: Z00A
Model: ASUS_Z00A
Id: MOB30D
Product: WW_Z00A

************ FIRMWARE ************
SDK: 23
Release: 6.0.1
Incremental: 06cfe67fe4

`
I hope it is of any help.

**Comment by davivel** [comment]
@domneu , @budinm1 , could you try with the just-released 2.1.1 version of the app? You will need to select the camera folder of your favorite camera app in the app settings.

**Comment by domneu** [comment]
I can.  I will try it Tuesday.

**Comment by domneu** [comment]
It did not work, but I am willing to send somebody a Kyocera Brigadier to
help troubleshoot this problem.

Would that help?

Dominic

**Comment by davivel** [comment]
So, you still have a crash with version 2.1.1, or the bug / error message / result is different?

In any case, could you please test this other APK? : https://github.com/owncloud/android/raw/instant_upload_combined_solution/android_2.1.2-beta-debug.apk

**Comment by domneu** [comment]
Sorry, just able to try this now, and the link doesn't work anymore.  But,
what happens is really nothing.  I have the settings correct, take a
picture, but nothing happens.  No OC upload in the notification area at the
top of the screen, no errors....

Maybe this helps?  Just point me to another link and I can try it tomorrow
morning.

Thank you for all of your help on this!

Dominic

**Comment by davivel** [comment]
@domneu , release 2.1.2 of the app is now available in F-Droid: https://f-droid.org/repository/browse/?fdfilter=owncloud&fdid=com.owncloud.android

Please, try it with that.

**Comment by davivel** [comment]
Beta process is going to change. Current beta branch will not be maintained anymore. If this is detected in a stable version, please reopen.

And thanks for reporting! :)
